tests/test_views_item_put.py

```diff
@@ -36,15 +36,7 @@ def test_valid_put(app, es, test_records, content_type, search_url,
     with app.test_client() as client:
         url = record_url(pid)
         res = client.put(url, data=json.dumps(record.dumps()), headers=HEADERS)
-        assert res.status_code == 200
-
-        # Check that the returned record matches the given data
-        assert get_json(res)['metadata']['year'] == 1234
-        IndexFlusher(search_class).flush_and_wait()
-        res = client.get(search_url, query_string={"year": 1234})
-        assert_hits_len(res, 1)
-        # Retrieve record via get request
-        assert get_json(client.get(url))['metadata']['year'] == 1234
+        assert res.status_code == 400
```

Comment:
do you want to test the invalid or valid case here? name is `test_valid_put`, but you're testing the case when the header is not present. I think you probably want to test both.

Reply:
I'm testing both cases, 👍 the naming of the test was wrong.

Reply:
where's the valid put test now?

Reply:
https://github.com/inspirehep/invenio-records-rest/pull/10/files#diff-a8354eebb456ed950c17919d31071af0b95e5f6ca65e55f917635955d1ed0018L53

Reply:
link doesn't work for me

Reply:
oh it's a mess with github view, it's the next test if you expand the code